### PR TITLE
OSD-8075 - Enforce webhooks via generated namespace lists

### PR DESCRIPTION
**Pull request by tnierman**
(Hopefully) completes: https://issues.redhat.com/browse/OSD-8075

---

Follow up to #168 and #166 

Adds a `Makefile` rule to generate a `PrivilegedNamespaces` list from the relevant configMaps in managed-cluster-config.

Webhooks will be enforced using this list, rather than using regular expressions covering all `openshift-.*|kube-.*|redhat-.*` namespaces.

### CODE REVIEW

**Comment by jharrington22** [comment]
/lgtm

@csheremeta @robotmaxtron @rogbas I'd like to see this merged today if you could approve it :)

File: pkg/config/generate/namespaces.go [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+}
+
+var (
+	// Base lists - default values which will always be enforced regardless of managed-cluster-config
+	namespaces = []string{"^redhat.*"}
+	configmaps = []string{}
+)
+
+const (
+	// generatedFileName defines the path to the generated file relative to the invoking go:generate command
```
Comment (lisa) [inline pkg/config/generate/namespaces.go:32]: Great comment 👍

**Review by jharrington22 — APPROVED** [review]
/lgtm

**Review by lisa — APPROVED** [review]
/lgtm

One thing to look into maybe is some Github syntactic sugar to consider the generated files as auto-generated, so they don't show up in the diff list. It's really outside the scope of the change (since these are obviously valid Go files and no problem with the filenames), but 🤷‍♀️ it'll help future reviews.

**Comment by jharrington22** [comment]
Manually approving this PR, its well tested and reviewed.

**Pull request merged**